### PR TITLE
Mention discord users in messages sent from IRC

### LINKED ISSUE #30: Mentions from IRC -> Discord
I am using a framework for my Discord Bot, and it has a regex to autoconvert `@username` to a mention. Passing the code for your ease of implementation.

```js
.replace(/@[^\s]+\b/g, (o) => {
    var member = message.channel.server.members.get("username", o.substring(1));
    return member ? member.mention() : o;
});
```

### PR DESCRIPTION
**Pull request by rce**
This was requested by my friends. Also closes #30.

Hopefully I'll have the time to get more contributions coming!

### CODE REVIEW

File: lib/bot.js [hunk 1]
```diff
@@ -163,8 +163,13 @@ class Bot {
         return;
       }
 
+      const withMentions = text.replace(/@[^\s]+\b/g, m => {
+        const user = this.discord.users.get('username', m.substring(1));
+        return user ? user.mention() : 0;
```
Comment (ekmartin) [inline lib/bot.js]: Shouldn't this be `return user ? user.mention () : m`? I.e. not `0`. Also, could probably rename m to `match` or something more readable.

**Comment by ekmartin** [comment]
Great! Do you think you could write a couple of tests for this too? Feel free to ask here or PM me on Discord (username ekmartin, server link here: http://www.reactiflux.com/) if you have any questions.

By the way, over at slack-irc (the application this is based on) we try to prefix usernames with an `@` if they're valid Slack usernames - that is, they're a member of the Slack channel the message will be sent to. Relevant pull request: https://github.com/ekmartin/slack-irc/pull/68

Updated code: https://github.com/ekmartin/slack-irc/blob/master/lib%2Fhelpers.js#L30-L51
https://github.com/ekmartin/slack-irc/blob/master/lib%2Fbot.js#L217-L223

Could be something to think about for another pull request later on. This is more important first though, what I'm mentioning here is more of a nice to have.

**Comment by rce** [comment]
@ekmartin I added some tests! Hopefully they are fine.

I may look into adding `@`-prefixes later.

File: test/stubs/discord-stub.js [hunk 2]
```diff
@@ -12,6 +12,9 @@ export function getChannel(key, value) {
 class DiscordStub extends events.EventEmitter {
   constructor() {
     super();
+    this.users = {
+      get: sinon.stub()
```
Comment (ekmartin) [inline test/stubs/discord-stub.js]: Maybe this should be a sandbox stub, like the ones defined here? https://github.com/rce/discord-irc/blob/irc-to-discord-mentions/test/bot.test.js#L29 That way it gets reset after each test, here: https://github.com/rce/discord-irc/blob/irc-to-discord-mentions/test/bot.test.js#L38

**Comment by ekmartin** [comment]
Other than my comment it looks great! :)

**Comment by rce** [comment]
@ekmartin Using `sandbox.stub()` now!

**Comment by ekmartin** [comment]
Great, thanks!

**Comment by ekmartin** [comment]
Published in 0.6.0 :)

**Pull request merged**